// Copyright 2018 Xcalar, Inc. All rights reserved.
//
// No use, or distribution, of this source code is permitted in any form or
// means without a valid, written license agreement with Xcalar, Inc.
// Please refer to the included "COPYING" file for terms and conditions
// regarding the use and redistribution of this software.
//

syntax = "proto3";

import "xcalar/compute/localtypes/Workbook.proto";

package xcalar.compute.localtypes.UDF;

message UdfModule {
    xcalar.compute.localtypes.Workbook.WorkbookScope scope = 1;
    string name = 2;
};

message FQname {
    string text = 1;
};

message GetResolutionRequest {
    UdfModule udfModule = 1;
};

message GetResolutionResponse {
    FQname fqModName = 1;
};

service UserDefinedFunction {
    rpc GetResolution (GetResolutionRequest) returns (GetResolutionResponse);
};
